src/Sylius/Bundle/CoreBundle/spec/EventListener/VariantOnhandListenerSpec.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\CoreBundle\EventListener;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachineInterface;
+use Sylius\Bundle\CoreBundle\EventListener\VariantOnhandListener;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Inventory\InventoryUnitTransitions;
+use Sylius\Component\Inventory\Model\InventoryUnitInterface;
+use Sylius\Component\Inventory\Repository\InventoryUnitRepositoryInterface;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * @author Robin Jansen <robinjansen51@gmail.com>
+ */
+class VariantOnhandListenerSpec extends ObjectBehavior
+{
+    function let(
+        FactoryInterface $factory,
+        InventoryUnitRepositoryInterface $inventoryUnitRepository
+    ) {
+        $this->beConstructedWith($factory, $inventoryUnitRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(VariantOnhandListener::class);
+    }
+
+    function it_fills_backordered_units_and_set_rest_to_onhand(
+        $factory,
+        $inventoryUnitRepository,
+        GenericEvent $event,
+        ProductVariantInterface $variant,
+        StateMachineInterface $stateMachine,
+        OrderInterface $order1,
+        OrderInterface $order2,
+        OrderItemInterface $orderItem1,
+        OrderItemInterface $orderItem2,
+        OrderItemUnitInterface $orderItemUnit1,
+        OrderItemUnitInterface $orderItemUnit2
+    ) {
+        $event->getSubject()->shouldBeCalled()->willReturn($variant);
+
+        $variant->getOnHand()->shouldBeCalled()->willReturn(5);
```

Comment:
It is better to use one of `shouldBeCalled` or `willReturn`, but rather not both. Only exception here is builder patter, which force us to use both of them